### PR TITLE
Store Promotions: Add selectors for edit states

### PR DESCRIPTION
**Pull request by coderkevin**
This adds selectors to get promotion edits, edits overlaid on top of the
existing promotions, and the currently editing promotion.

To Test: `npm test`

### CODE REVIEW

File: client/extensions/woocommerce/state/selectors/promotions.js [hunk 1]
```diff
@@ -1,31 +1,60 @@
...
+export function getCurrentlyEditingPromotionId( rootState, siteId = getSelectedSiteWithFallback( rootState ) ) {
+	const edits = get( rootState, [ 'extensions', 'woocommerce', 'ui', 'promotions', 'edits', siteId ], {} );
+	return edits.currentlyEditingId || null;
+}
+
+export function getPromotionEdits( rootState, promotionId, siteId = getSelectedSiteWithFallback( rootState ) ) {
+	const edits = get( rootState, [ 'extensions', 'woocommerce', 'ui', 'promotions', 'edits', siteId ], {} );
+
+	if ( isObject( promotionId ) ) {
+		return ( edits.creates && edits.creates.find( ( p ) => promotionId === p.id ) ) || null;
```
Comment (timmyc) [inline client/extensions/woocommerce/state/selectors/promotions.js]: if `promotionId` is an object, will/can this clause ever be true? looking at the test cases, it seems like `promotionId` is only ever tested as a string value - do we need other test cases here or should this just be `if promotionId`?

File: client/extensions/woocommerce/state/selectors/promotions.js [hunk 2]
```diff
@@ -1,31 +1,60 @@
...
 
 export function getPromotions( rootState, siteId = getSelectedSiteWithFallback( rootState ) ) {
-	const { promotions } = get( rootState, [ 'extensions', 'woocommerce', 'sites', siteId ], {} );
+	const promotions = get( rootState, [ 'extensions', 'woocommerce', 'sites', siteId, 'promotions' ], {} );
 	return promotions.promotions;
 }
 
+export function getPromotion( rootState, promotionId, siteId = getSelectedSiteWithFallback( rootState ) ) {
+	const promotions = getPromotions( rootState, siteId );
+	return promotions.find( ( p ) => promotionId === p.id ) || null;
```
Comment (allendav) [inline client/extensions/woocommerce/state/selectors/promotions.js]: FWIW, I'm a fan of lodash's find instead - I find it easier to quickly grok, e.g.

`return find( promotions, { id: promotionId } ) || null`

File: client/extensions/woocommerce/state/selectors/promotions.js (hunk 1 above)
Comment (allendav) [inline client/extensions/woocommerce/state/selectors/promotions.js]: Same comment above about a personal preference for lodash find. YMMV.

File: client/extensions/woocommerce/state/selectors/test/promotions.js [hunk 3]
```diff
@@ -29,9 +34,9 @@ describe( 'promotions', () => {
 					123: {
 						promotions: {
 							promotions: [
-								{ type: 'empty1' },
-								{ type: 'empty2' },
-								{ type: 'empty3' },
+								{ id: 'id1', type: 'empty1' },
```
Comment (allendav) [inline client/extensions/woocommerce/state/selectors/test/promotions.js]: Suggestion - your promotion IDs in the real world will look like coupon:99 or product:86, right? maybe use IDs like that in your test cases too.

File: client/extensions/woocommerce/state/selectors/promotions.js (hunk 1 above)
Comment (timmyc) [reply client/extensions/woocommerce/state/selectors/promotions.js]: I believe this is only a placeholder on the initial `create` call, so is kind of a transient - at least that is how I read it.

Comment (coderkevin) [reply client/extensions/woocommerce/state/selectors/promotions.js]: Good advice. I added a test for this. Thanks.

File: client/extensions/woocommerce/state/selectors/promotions.js (hunk 2 above)
Comment (coderkevin) [reply client/extensions/woocommerce/state/selectors/promotions.js]: I've grown to like lodash's find lately as well. I changed the code to use it instead.

File: client/extensions/woocommerce/state/selectors/test/promotions.js (hunk 3 above)
Comment (coderkevin) [reply client/extensions/woocommerce/state/selectors/test/promotions.js]: Changed the ids to coupon:1, product:2, coupon:3

**Comment by coderkevin** [comment]
Adjusted test code, etc. Rebased.

**Comment by coderkevin** [comment]
Rebased on top of previous PR + new `origin/master`, which included the recent formatting/Prettier updates.

**Comment by coderkevin** [comment]
Merged 1 other PR into this before merging to master.

**Pull request merged**